### ISSUE TITLE
Ressourcewarning: unclosed file in UnitTest

### ISSUE THREAD

**Issue by b-m-f** [gabrielfalcao/HTTPretty#368]
This is my test:
```
import httpretty
import unittest
import requests
from unittest import TestCase

file_cache_directory = ".testcache"

class Test(TestCase):
    @httpretty.activate
    def test(self):
        httpretty.register_uri(
            httpretty.GET,
            "https://httpbin.org/ip",
            body='{"origin": "127.0.0.1"}',
        )
        resp = requests.get("https://httpbin.org/ip")

        self.assertIn("127.0.0.1", resp.text)
```

Which generates the following error while still executing successfully:

```
httpretty/core.py:495: ResourceWarning: unclosed file <_io.BufferedRandom name=4>
  self.fd = FakeSockFile()
```

**HTTPretty VERSION**:  0.9.6
**Python VERSION**:  3.6.7
**OS**: 64bit Linux

**Comment by fraank** [comment]
Not a solution, but a quick fix is to add this in your setUp-Method:

    def setUp(self):
        warnings.filterwarnings("ignore", category=ResourceWarning, message="unclosed.*

Do not forget to..

    import warnings

**Comment by Brad-eki** [comment]
Have bumped into this. Following.

**Comment by gabrielfalcao** [comment]
closed by #371
